subprojects/plugins/src/main/java/org/gradle/api/tasks/SourceSetOutput.java

```diff
@@ -54,33 +54,30 @@
  *
  * <pre class='autoTested'>
  * plugins {
- *     id 'java'
+ *   id 'java'
  * }
  *
- * def generatedResources = "$buildDir/generated-resources/main"
+ * def generateResourcesTask = tasks.register("generate-resources", GenerateResourcesTask) {
+ *   resourcesDir.set(file("$buildDir/generated-resources/main"))
+ * }
  *
  * sourceSets {
  *   main {
- *     //let's register an output folder on the main SourceSet:
- *     output.dir(generatedResources, builtBy: 'generateMyResources')
- *     //it is now a part of the 'main' classpath and will be a part of the jar
+ *     // Register the generated 'resourcesDir' as an output of the main source set, retaining the task dependency.
+ *     output.dir(generateResourcesTask.flatMap {it.resourcesDir} )
```

Comment:
Does this work if you just leave it as `generateResourcesTask`?

Reply:
Yes it does. Updated.